dvc/scm/git.py

```diff
@@ -71,6 +73,7 @@ def _get_gitignore(self, path):
         return entry, gitignore
 
     def ignore(self, path):
+        self.ignored_paths.append(path)
```

Comment:
Let's add it to the list _after_ we've added it to gitignore itself. E.g. it might be that we already have an entry for it, so we shouldn't remove what we didn't add. Looks like it is worth adding a test for it.